src/components/package-port/CodeAndPreview.tsx

```diff
@@ -346,7 +308,16 @@ export function CodeAndPreview({ pkg }: Props) {
     })
   }
 
-  const { handleCreateFile } = useFileManagement(state, setState)
+  const { handleCreateFile, handleDeleteFile, handleSave, fsMap } =
+    useFileManagement({
+      setCodeAndPreviewState: setState,
+      pkg,
+      updatePackageFilesMutation,
+      openNewPackageSaveDialog,
+      refetchPackageFiles: pkgFiles.refetch,
```

Comment:
it should be fetching the files itself, they shouldn't be passed in. You can use `invalidateQueries` to cause package file invalidation